### PR TITLE
Allow for back link on top of form + enable for 4138

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- Allow for back link on top of form + enable for 4138
- Add redux state `form.layout` and `form.layout.useTopBackLink`

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team-forms#1263
department-of-veterans-affairs/va.gov-team-forms#1266

## Testing done

- Browser testing

## Screenshots

image
image

## What areas of the site does it impact?

4138, and ability to use on other forms.

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/forms-system/src/js/containers/FormApp.jsx [hunk 1]
```diff
@@ -3,16 +3,28 @@ import { connect } from 'react-redux';
...
 
 const { Element } = Scroll;
 
+const BeforePage = ({ useTopBackLink }) => {
+  return useTopBackLink ? <BackLink /> : null;
+};
+
+const BeforePageContent = connect(state => ({
+  useTopBackLink: state?.form?.layout?.useTopBackLink,
+}))(BeforePage);
```
Comment (Mottie) [inline src/platform/forms-system/src/js/containers/FormApp.jsx:22]: I've never seen `connect` used within a component. We are using a new enough version of redux that includes [`useSelector`](https://react-redux.js.org/api/hooks#useselector). If we use a form global `useTopBackLink`, FormApp does have access to the `formConfig`. If you want to keep it per page then I think we can use a combo of `currentLocation` and `pageList` to find the page config.

Comment (rhasselle-oddball) [reply src/platform/forms-system/src/js/containers/FormApp.jsx:22]: Great callout - relying on formConfig seems like a good alternate, and can scrap redux altogether.

**Review by Mottie — APPROVED** [review]
Looks great! Thanks for working on this 🎉

**Review by CBonade — APPROVED** [review]

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

**Pull request merged**